File: practice/algorithms/implementation/caesar_cipher/CaesarCipher.scala
import scala.io.Source

object CaesarCipher extends App {
  val lines = Source.stdin.getLines().toList
  val text = lines(1)
  val encryptionKey = lines(2).toInt
  val encrypted = text.map(caesarCipher(_, encryptionKey))
  println(encrypted)

  def caesarCipher(c: Char, shift: Int): Char = {
    if(c >= 'a' && c <= 'z') {
      return ((c.toInt - 'a'.toInt + shift) % 26 + 'a'.toInt).toChar
    }
    if(c>= 'A' && c <= 'Z') {
      return ((c.toInt - 'A'.toInt + shift) % 26 + 'A'.toInt).toChar
    }
    return c
  }
}

File: practice/algorithms/implementation/caesar_cipher/CaesarCipher.java
import java.util.*;

public class CaesarCipher {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int length = stdin.nextInt();
        String text = stdin.next();
        int shift = stdin.nextInt();
        for(int i = 0; i < length; i++) {
            char c = text.charAt(i);
            if(c >= 'a' && c <= 'z') {
                System.out.print((char)(((int)c - (int)'a' + shift) % 26 + (int)'a'));
            } else if(c >= 'A' && c <= 'Z') {
                System.out.print((char)(((int)c - (int)'A' + shift) % 26 + (int)'A'));
            } else {
                System.out.print(c);
            }
        }
        stdin.close();
    }
}